homeassistant/components/overkiz/water_heater_entities/atlantic_dhw.py

```diff
@@ -0,0 +1,164 @@
+"""Support for Atlantic Pass APC DHW."""
+
+import logging
+from typing import Any, cast
+
+from pyoverkiz.enums import OverkizCommand, OverkizCommandParam, OverkizState
+
+from homeassistant.components.water_heater import (
+    STATE_ECO,
+    STATE_OFF,
+    STATE_PERFORMANCE,
+    WaterHeaterEntity,
+    WaterHeaterEntityFeature,
+)
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+
+from ..entity import OverkizEntity
+
+OVERKIZ_TO_OPERATION_MODE: dict[str, str] = {
+    STATE_PERFORMANCE: STATE_PERFORMANCE,
+    STATE_ECO: OverkizCommandParam.MANUAL_ECO_ACTIVE,
+    OverkizCommandParam.MANUAL: OverkizCommandParam.MANUAL_ECO_INACTIVE,
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AtlanticDHW(OverkizEntity, WaterHeaterEntity):
+    """Representation of Atlantic Water Heater."""
+
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_supported_features = (
+        WaterHeaterEntityFeature.TARGET_TEMPERATURE
+        | WaterHeaterEntityFeature.OPERATION_MODE
+        | WaterHeaterEntityFeature.AWAY_MODE
+        | WaterHeaterEntityFeature.ON_OFF
+    )
+    _attr_operation_list = [*OVERKIZ_TO_OPERATION_MODE.keys()]
+    _attr_min_temp = 50.0
```

Comment:
Maybe you could use those states for the min/max temps:
```
core:MinimalTemperatureManualModeState
core:MaximalTemperatureManualModeState
```

Reply:
I tried making _attr_min_temp and _attr_max_temp as properties with those attribute getters
```python
    @property
    def _attr_max_temp(self) -> float:
        """Return the current temperature."""
        return cast(
            float,
            self.executor.select_state(
                OverkizState.CORE_MAXIMAL_TEMPERATURE_MANUAL_MODE
            ),
        )

    
    @property
    def _attr_min_temp(self) -> float:
        """Return the current temperature."""
        return cast(
            float,
            self.executor.select_state(
                OverkizState.CORE_MINIMAL_TEMPERATURE_MANUAL_MODE
            ),
        )
```
but pre-commit told me that I cannot do this :\
I looked at other examples and these fields are always constants. I don't know why...